File: src/main/scala/jk_5/nailed/server/map/DummyLobbyMappack.scala
/*
 * Nailed, a Minecraft PvP server framework
 * Copyright (C) jk-5 <http://github.com/jk-5/>
 * Copyright (C) Nailed team and contributors <http://github.com/nailed/>
 *
 * This program is free software: you can redistribute it and/or modify it
 * under the terms of the MIT License.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License
 * for more details.
 *
 * You should have received a copy of the MIT License along with
 * this program. If not, see <http://opensource.org/licenses/MIT/>.
 */

package jk_5.nailed.server.map

import java.io.File

import io.netty.util.concurrent.Promise
import jk_5.nailed.api.mappack.Mappack

/**
 * No description given
 *
 * @author jk-5
 */
object DummyLobbyMappack extends Mappack {
  override val getId = "nailed:lobby"
  override def getMetadata = DummyLobbyMappackMetadata
  override def prepareWorld(destinationDirectory: File, promise: Promise[Void]){
    promise.setSuccess(null)
  }
}

File: src/main/java/jk_5/nailed/server/map/DummyLobbyMappack.java
package jk_5.nailed.server.map;

import io.netty.util.concurrent.Promise;
import jk_5.nailed.api.mappack.Mappack;
import jk_5.nailed.api.mappack.filesystem.IMount;
import jk_5.nailed.api.mappack.metadata.MappackMetadata;
import jk_5.nailed.server.mappack.metadata.xml.XmlMappackMetadata;

import javax.annotation.Nonnull;
import javax.annotation.Nullable;
import java.io.File;

public class DummyLobbyMappack implements Mappack {

    private static final DummyLobbyMappack INSTANCE = new DummyLobbyMappack();
    private final MappackMetadata metadata;

    public DummyLobbyMappack() {
        try{
            this.metadata = XmlMappackMetadata.fromResource("dummy-mappack.xml");
        }catch(Exception e){
            throw new RuntimeException(e);
        }
    }

    @Nonnull
    @Override
    public String getId() {
        return "nailed/lobby";
    }

    @Nonnull
    @Override
    public MappackMetadata getMetadata() {
        return metadata;
    }

    @Override
    public void prepareWorld(@Nonnull File destinationDirectory, @Nonnull Promise<Void> promise) {
        promise.setSuccess(null);
    }

    @Nullable
    @Override
    public IMount getMappackMount() {
        return null;
    }

    public static Mappack instance(){
        return INSTANCE;
    }
}
